Consolidate some Mach lookup sandbox rules

pvollan opened the pull request:
#### 8bf99a30f0cbdb266ae28bc15ed6ff4102ca13f4
<pre>
Consolidate some Mach lookup sandbox rules
<a href="https://bugs.webkit.org/show_bug.cgi?id=297825">https://bugs.webkit.org/show_bug.cgi?id=297825</a>
<a href="https://rdar.apple.com/159014309">rdar://159014309</a>

Reviewed by Brent Fulgham.

For clarity, consolidate some Mach lookup sandbox rules in the WebContent process on iOS.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: <a href="https://commits.webkit.org/299156@main">https://commits.webkit.org/299156@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04bb9a6bd23cbae203ad0dd1a2d67f885b48468c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117797 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37475 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28105 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123933 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69820 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1cc3efd1-b7d4-4a70-acdd-6671eab4d445) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38167 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46057 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89400 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59034 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/75054d48-5b47-4419-b6bd-243f0413e8ea) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120749 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30387 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105621 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69893 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4ba9ceb5-1e60-4ba0-98bf-500ff8a9e6b6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29451 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67597 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99807 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23916 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127024 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44700 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33652 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98061 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45058 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101846 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97849 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43232 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21201 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41072 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18815 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44572 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50246 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44030 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47377 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45721 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -752,21 +753,6 @@
-        (require-all
-            (require-not (webcontent-process-launched))
-            (extension "com.apple.webkit.extension.mach")
-            (global-name "com.apple.mobilegestalt.xpc"))))
```
Did you intend to remove this extension-controlled access to MobileGestalt? Or maybe this is just dead code if all sandboxes are known to be `sandbox-version-2` or newer?

pvollan replied:
That is a good point! Yes, this is dead code now.

Thanks for reviewing!

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -819,19 +805,6 @@
-        (require-all
-            (extension "com.apple.webkit.extension.mach")
-            (system-attribute apple-internal)
-            (global-name "com.apple.webinspector"))))
```
Likewise: Do we no longer need this extension-controlled WebInspector connection? Or are all sandboxes now known to be `sandbox-version-2` (or newer) so this is just dead code?

pvollan replied:
Yes, this is also dead code now.

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -401,11 +401,24 @@
+        "com.apple.SystemConfiguration.configd"
+        "com.apple.aggregated"
+        "com.apple.analyticsd"
+        "com.apple.audio.AudioComponentRegistrar"
```
Previously we only denied `AudioComponentRegistrar` if we handn't been granted an extension. Is this correct now?

pvollan replied:
Yes, this should be correct, since the default is to deny Mach lookup. So the extra rule for `AudioComponentRegistrar` with an extra condition was redundant.

brentfulgham reviewed (approved):
LGTM, but please consider my questions in case they were oversights.

pvollan committed:
Consolidate some Mach lookup sandbox rules
https://bugs.webkit.org/show_bug.cgi?id=297825
rdar://159014309

Reviewed by Brent Fulgham.

For clarity, consolidate some Mach lookup sandbox rules in the WebContent process on iOS.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: https://commits.webkit.org/299156@main
